util/buildkitd/buildkitd.go

```diff
@@ -59,8 +68,32 @@ func Start(ctx context.Context) (string, error) {
 func checkBuildkit(ctx context.Context) error {
 	lg := log.Ctx(ctx)
 
+	lockFilePath, err := homedir.Expand(buildkitdLockPath)
+	if err != nil {
+		return fmt.Errorf("unable to expand buildkitd lock path: %w", err)
+	}
+	if err := os.MkdirAll(filepath.Dir(lockFilePath), 0755); err != nil {
+		return fmt.Errorf("unable to create buildkitd lock path parent dir: %w", err)
+	}
+	lock := flock.New(lockFilePath)
+	lockCtx, cancel := context.WithTimeout(ctx, lockTimeout)
+	defer cancel()
+	locked, err := lock.TryLockContext(lockCtx, 100*time.Millisecond)
```

Comment:
I'm wondering if we should log anything as `dagger` is waiting on this lock (so people aren't confused by the fact that `dagger` might just sit there for a decent bit of time waiting to get the lock. 

Having said that, this shouldn't arise _too_ often for users. 

Beyond that, looks good to me!

Reply:
Good idea, updated with some debug logs.